daemon/daemon_unix.go

```diff
@@ -375,6 +390,13 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 		warnings = append(warnings, "IPv4 forwarding is disabled. Networking will not work.")
 		logrus.Warnf("IPv4 forwarding is disabled. Networking will not work")
 	}
+	if hostConfig.CgroupParent != "" && daemon.usingSystemd() {
+		// CgroupParent for systemd cgroup should be named as "xxx.slice"
+		length := len(hostConfig.CgroupParent)
+		if length <= 6 || hostConfig.CgroupParent[length-6:] != ".slice" {
```

Comment:
Well, doing this systemd slice stuff only really makes sense if you're actually guaranteed to be referencing slices -- which is only the case if you have something like `--cgroup-parent=/test.slice`. If you use `--cgroup-parent=test.slice`, you won't get what you expect (we were burned by this with SLE).